core/deployment/src/main/java/io/quarkus/deployment/steps/BannerProcessor.java

```diff
@@ -0,0 +1,122 @@
+package io.quarkus.deployment.steps;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.banner.BannerConfig;
+import io.quarkus.builder.Version;
+import io.quarkus.deployment.IsTest;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ConsoleFormatterBannerBuildItem;
+import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
+import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.runtime.BannerRecorder;
+import io.quarkus.runtime.BannerRuntimeConfig;
+
+public class BannerProcessor {
+
+    private static final Logger logger = Logger.getLogger(BannerProcessor.class);
+
+    @BuildStep(loadsApplicationClasses = true, onlyIfNot = { IsTest.class })
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public ConsoleFormatterBannerBuildItem recordBanner(BannerRecorder recorder, BannerConfig config,
+            BannerRuntimeConfig bannerRuntimeConfig) {
+        String bannerText = readBannerFile(config);
+        return new ConsoleFormatterBannerBuildItem(recorder.provideBannerSupplier(bannerText, bannerRuntimeConfig));
+    }
+
+    @BuildStep
+    HotDeploymentWatchedFileBuildItem watchBannerChanges(BannerConfig config) {
+        return new HotDeploymentWatchedFileBuildItem(config.path);
+    }
+
+    private String readBannerFile(BannerConfig config) {
+        try {
+            Map.Entry<URL, Boolean> entry = getBanner(config);
+            URL bannerResourceURL = entry.getKey();
+            if (bannerResourceURL == null) {
+                logger.warn("Could not locate banner file");
+                return "";
+            }
+            try (InputStream is = bannerResourceURL.openStream()) {
+                byte[] content = FileUtil.readFileContents(is);
+                String bannerTitle = new String(content, StandardCharsets.UTF_8);
+
+                int width = 0;
+                Scanner scanner = new Scanner(bannerTitle);
+                while (scanner.hasNextLine()) {
+                    width = Math.max(width, scanner.nextLine().length());
+                }
+
+                String tagline = "\n";
+                Boolean isDefaultBanner = entry.getValue();
+                if (!isDefaultBanner) {
```

Comment:
The more and more I think about it i think we are better of just always print this tag line...

Reply:
I thought @emmanuelbernard wanted to always print this tag line?

Reply:
All I said is that it said  `Quarkus` in the banner and right below `powered by Quarkus v123` and Quarkus powered by Quarkus felt wrong to me.

But since we put the version too, maybe that's fine and we just always say `Powered by Quarkus v123`. I certainly did not want to spend 5FTE for 5 years on this problem :D

Reply:
In that spirit, I saw we merge it now and then people can test and improve on something that actually exists and isn't vaporware :)